app/controllers/v1/tasks_controller.rb

```diff
@@ -0,0 +1,20 @@
+module V1
+  class TasksController < ApplicationController
+    swagger_controller :tasks, 'Tasks'
+
+    swagger_api :update do
+      summary 'Updates an existing Task'
+      param :task_id, :integer, :required, 'Task Id'
+    end
+    def update_task_state
```

Comment:
I can see how that would work. You could use the start endpoint to update the task from “todo” to “in-progress” and then when you were done with a task you’d mark it as complete and by doing so call the finished endpoint. That would update the task to done and send the sms message. I think your suggestion is better in the fact that it separates the logic that handles whether to send an sms message. Separation of concerns 👍 

This also simplifies the model by removing the _send_completion_sms?_ method I created, drying up the code.

**projects_controller.rb**
```
def start
  task = Task.find_by(id: params[:task_id])
  task.update(state: "in-progress")
   
  render json: task, serializer: TaskSerializer, root: false
end


def finish
  task = Task.find_by(id: params[:task_id])
  task.update(state: "done")
  save #I know you wrote below that self-saving methods tend to be no bueno, 
 
  message = “Hooray! Task Complete!”
  task.send_message_via_sms(message)
 
  render json: { completed: task }, root: false
end
```